snapcraft/internal/cache/_snap.py

```diff
@@ -45,7 +45,8 @@ def cache(self, snap_filename, revision):
             revision)
         cached_snap_path = os.path.join(self.snap_cache_dir, cached_snap)
         try:
-            link_or_copy(snap_filename, cached_snap_path)
+            if not os.path.isfile(cached_snap_path):
+                shutil.copyfile(snap_filename, cached_snap_path)
```

Comment:
Please add a code comment on why copyfile so the next person coming in doesn't "optimize" this :wink: